MELOSYS-3414 - frontend som egen pod

sigbjorn-myhre opened the pull request:
Legger til oppsett for å kunne deploye uten backend. Bruker en [nginx-container](https://github.com/navikt/nginx-oidc-docker) som tar seg av proxy til backend eller til isso, og som sender ID_token videre til backend. Dette oppsettet forventer en [redis-pod.](https://github.com/navikt/melosys-web/blob/c0b3ca8651033f479b7b6a9c3c47f29d82107ade/nais/t8/redis.yaml) for å lagre sesjon.

[Nginx proxier](https://github.com/navikt/melosys-web/blob/c0b3ca8651033f479b7b6a9c3c47f29d82107ade/nais/proxy.nginx) all trafikk med stien `/api/*` til backend. Det er også en reverse proxy som redirecter alt til `/melosys` (via isso). I [routing.js](https://github.com/navikt/melosys-web/pull/810/files#diff-e3b846730602d9f7e7a98ca075410c99) har jeg lagt inn redirect tilbake til `/`, slik at stiene i applikasjonen fremstår uendret.

Har også lagt til en action som deployer til t8 ved push til en sprint-branch. `frontendlogger/logger.js` er nå lagt til som en statisk fil i frontend (i stedet for å hente den fra backend).

sigbjorn-myhre committed:
enkel node-server med azuread-config

sigbjorn-myhre committed:
legger til config for å deploye med nginx

sigbjorn-myhre committed:
legger til frontendlogger i melosys-web

sigbjorn-myhre committed:
Oppdater dockerfile og legg til filer for deploy til nais

sigbjorn-myhre committed:
sett rett homepage

sigbjorn-myhre committed:
Merge branch 'SPRINT-40' into MELOSYS-3414_frontend_som_egen_pod

sigbjorn-myhre committed:
Legger til jobb for å deploye til t8

Commit:
Forsøk contain()

sigbjorn-myhre committed:
Mangler GITHUB_TOKEN ved docker login

sigbjorn-myhre committed:
Bare deploy ved push til sprint-brancher

AStrand94 commented on .github/workflows/push.yml:
```diff
@@ -0,0 +1,90 @@
+      - name: Push Docker image
+        run: docker push $(cat .docker_image):$(cat .docker_tag)
+
+      - name: Deploy til t8
```
Hadde vært fornuftig med en Slack-melding som i Melosys.

sigbjorn-myhre replied:
Enig. Jeg legger det til.

JohanSakshaug commented on .github/workflows/push.yml:
```diff
@@ -0,0 +1,90 @@
+      - name: build
+        run: npm run build
+
+      - name: Upload bundle
```
Kunne det vært en ide å bare laste opp bundle hvis vi er på SPRINT-*? Vi vil jo ikke trenge den hvis vi ikke skal deploye den.
https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstepsif

AStrand94 commented on .github/workflows/push.yml:
```diff
@@ -0,0 +1,90 @@
+          NAME: melosys-web
+        run: |
+          echo "docker.pkg.github.com"/"$GITHUB_REPOSITORY"/"$NAME" > .docker_image
+          echo "$(node -p "require('./package.json').version")-$(git rev-parse --short HEAD)" > .docker_tag
```
Jeg synes versjonering er unødvendig når melosys-web blir sin egen app. Jeg synes man kan gå over til format 'dato-commithash'.
`echo "$(date "+%Y.%m.%d")-$(git rev-parse --short HEAD)" > .docker_tag`

sigbjorn-myhre replied:
@JohanSakshaug tanker?

JohanSakshaug replied:
Som vi snakket om i går, jeg er enig.

JohanSakshaug commented on src/routing.js:
```diff
@@ -221,9 +221,12 @@ const Routing = ({
     visRevurderVedtakDialogHandle,
   };
 
+  const utenPrefix = props => ({ pathname: props.location.pathname.replace('/melosys', '') });
```
pathUtenPrefix?

sigbjorn-myhre committed:
Legger til sending av slack-melding etter deploy

Fikser versjons-variabler som ikke ble satt under bygg
Fikser til rett tidssone for build-time
Laster opp bundle kun dersom commit skal deployes

sigbjorn-myhre committed:
Setter branch_name rett

JohanSakshaug commented on .github/workflows/push.yml:
```diff
@@ -0,0 +1,113 @@
+
+      - name: Slack Notification (deploy success)
+        if: success()
+        uses: rtCamp/action-slack-notify@master
```
Kan vi bruke rtCamp/action-slack-notify@v1, for sikkerhets skyld? (master kan endres)

sigbjorn-myhre replied:
absolutt! 👍

JohanSakshaug reviewed (approved):
👍

sigbjorn-myhre committed:
acktion-slack-notify@master -> @v1.0.0

sigbjorn-myhre committed:
Deploy redis i github action
